pkg/runtime.v2/framework/plugins/jobset/jobset.go

```diff
@@ -74,29 +74,37 @@ func (j *JobSet) Build(ctx context.Context, info *runtime.Info, trainJob *kubefl
 	if !ok {
 		return nil, nil
 	}
-	jobSetBuilder := NewBuilder(client.ObjectKeyFromObject(trainJob), kubeflowv2.JobSetTemplateSpec{
-		ObjectMeta: metav1.ObjectMeta{
-			Labels:      info.Labels,
-			Annotations: info.Annotations,
-		},
-		Spec: raw.Spec,
-	})
+
+	var jobSetBuilder *Builder
+	oldJobSet := &jobsetv1alpha2.JobSet{}
+	if err := j.client.Get(ctx, client.ObjectKeyFromObject(trainJob), oldJobSet); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil, err
+		}
+		jobSetBuilder = NewBuilder(client.ObjectKeyFromObject(trainJob), kubeflowv2.JobSetTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels:      info.Labels,
+				Annotations: info.Annotations,
+			},
+			Spec: raw.Spec,
+		})
+		oldJobSet = nil
```

Comment:
Do you need this ?

Reply:
Yes, we need this for the `needsCreateOrUpdate` function.